archinstall/lib/user_interaction.py

```diff
@@ -99,18 +99,29 @@ def print_large_list(options, padding=5, margin_bottom=0, separator=': '):
 	return column, row
 
 
-def generic_multi_select(options, text="Select one or more of the options above (leave blank to continue): ", sort=True, default=None, allow_empty=False):
+def generic_multi_select(options, text="Select one or more of the options above (leave blank to continue): ", sort=False, default=None, allow_empty=False):
+	# For now, we check for list, but in future it's better to have support for dictionary
+	# (At the moment there are no cases of using dictionaries with this function)
+	if type(options) not in [list]:
+		log(f" * Generic multi-select doesn't support ({type(options)}) as type of options * ", fg='red')
+		log(" * If problem persists, please create an issue on https://github.com/archlinux/archinstall/issues * ", fg='yellow')
+		raise RequirementError("generic_multi_select() requires list as options.")
+	if not options:
+		log(f" * Generic multi-select didn't find any options to choose from * ", fg='red')
+		log(" * If problem persists, please create an issue on https://github.com/archlinux/archinstall/issues * ", fg='yellow')
+		raise RequirementError('generic_multi_select() requires at least one option to proceed.')
+	# After passing the checks, function continues to work
 	if sort:
 		options = sorted(options)
 
 	section = MiniCurses(get_terminal_width(), len(options))
 
 	selected_options = []
 
-	while True:
-		if len(selected_options) <= 0 and default and default in options:
-			selected_options.append(default)
+	if not selected_options and default in options:
+		selected_options.append(default)
```

Comment:
The reason why these are *inside* the loop, is to avoid confusion on empty selections resulting in the `default` value. This will graphically show that `default` has to be selected if nothing else is. Otherwise you could unmark everything, press enter, and something got selected anyway going against what's visible to the user during selection.